### ISSUE TITLE
`rust-iai-callgrind` fails to collect all metrics

### ISSUE THREAD

**Issue by OmarTawfik** [bencherdev/bencher#453]
`iai-callgrind` has many tools. By default, it only runs `CALLGRIND`, which reports the following six metrics:

```log
iai::benchmarks::test1
  Instructions:          3102359729|3102359161      (+0.00002%) [+1.00000x]
  L1 Hits:               4435753829|4435875893      (-0.00275%) [-1.00003x]
  L2 Hits:                  1199262|1076439         (+11.4101%) [+1.11410x]
  RAM Hits:                   25535|25449           (+0.33793%) [+1.00338x]
  Total read+write:      4436978626|4436977781      (+0.00002%) [+1.00000x]
  Estimated Cycles:      4442643864|4442148803      (+0.01114%) [+1.00011x]
```

But when the `DHAT` tool is additionally enabled, the following metrics are reported as well (and the output is split into two sections):

```log
iai::benchmarks::test1
  ======= CALLGRIND =======================================================
  Instructions:          3102359729|3102359729      (No change)
  L1 Hits:               4435921602|4435921602      (No change)
  L2 Hits:                  1031503|1031503         (No change)
  RAM Hits:                   25521|25521           (No change)
  Total read+write:      4436978626|4436978626      (No change)
  Estimated Cycles:      4441972352|4441972352      (No change)
  ======= DHAT ============================================================
  Total bytes:             26294939|26294939        (No change)
  Total blocks:             2328086|2328086         (No change)
  At t-gmax bytes:           933718|933718          (No change)
  At t-gmax blocks:           18344|18344           (No change)
  At t-end bytes:                 0|0               (No change)
  At t-end blocks:                0|0               (No change)
  Reads bytes:             47577425|47577425        (No change)
  Writes bytes:            37733810|37733810        (No change)
```

Currently they are completely ignored by `bencher run`. I think they need to be added to the list [here](https://github.com/bencherdev/bencher/blob/8be4539d865070820aa8dbc6d6ee763db2f6c02c/lib/bencher_adapter/src/adapters/rust/iai_callgrind.rs#L67-L98).

Additionally, the benchmark name is incorrectly picked up as ` ======= CALLGRIND =======================================================`, instead of `iai::benchmarks::test1`, which should be handled when fixing this.

Here is a Rust repro that enables this additional tool:

```rust
#[library_benchmark]
fn test1() {
    black_box(execute_something());
}

library_benchmark_group!(
    name = benchmarks;
    benchmarks = test1
);

main!(
    config = LibraryBenchmarkConfig::default().tool(Tool::new(ValgrindTool::DHAT));
    library_benchmark_groups = benchmarks
);
```

cc @epompeii

**Comment by epompeii** [comment]
Thank you for pointing this out.
When `iai-callgrind` support was added in https://github.com/bencherdev/bencher/pull/319 I don't think this was under consideration.

I'm open to a PR to add support for this. @OmarTawfik you are welcome to do so. Also tagging @JonasWanke in case he is interested.

Whoever implements this needs to make sure that the ASCII escaping is properly handled: https://github.com/bencherdev/bencher/issues/345

**Comment by OmarTawfik** [comment]
I'll take a stab at it. Thanks for the pointers!

**Comment by OmarTawfik** [comment]
Fixed by #461

**Comment by epompeii** [comment]
Completed in `v0.4.18`: https://bencher.dev/docs/reference/changelog/#v0418
